### PR TITLE
feature/v1-070/responsive my courses

### PR DESCRIPTION
**Pull request by Julia-Maliutsina**
1. Responsive design for My courses page.
2. Responsive design for common Course component.
3. Different display of common component on Courses list and My courses pages for mobile devices.

### CODE REVIEW

**Review by dapodev — DISMISSED** [review]

**Review by hauteballe — DISMISSED** [review]

File: frontend/src/components/Course/Course.tsx [hunk 1]
```diff
@@ -13,25 +15,60 @@ import {
...
   <CourseContainer container direction="column">
     <AboutCourseContainer>
       <ImageWrapper>
         <Image />
       </ImageWrapper>
-      <CourseTitle>{title}</CourseTitle>
-      <CourseDescriptionWrapper>
-        <CourseDescription>{shortifyCourseDescription(description)}</CourseDescription>
-      </CourseDescriptionWrapper>
+      <div style={{ flexGrow: '2' }}>
```
[the remark this answers is not in the document]
Comment (Julia-Maliutsina) [reply frontend/src/components/Course/Course.tsx]: fixed

File: frontend/src/components/Course/Course.tsx [hunk 2]
```diff
@@ -13,25 +15,61 @@ import {
...
-      </CourseDescriptionWrapper>
+      <CourseTextContainer>
+        <CourseTitle>{title}</CourseTitle>
+        <CourseDescriptionWrapper>
+          <CourseDescription>{shortifyCourseDescription(description)}</CourseDescription>
+        </CourseDescriptionWrapper>
+        <MobileCourseInfoBox>
+          <CourseInfo duration={duration} lessons={lessons} />
+        </MobileCourseInfoBox>
+        {status === 'completed' && (
```
Comment (k-krasilnikova) [inline frontend/src/components/Course/Course.tsx:62]: We should reuse as many strings as possible. It's not a good practice to fix in each comparison the same string!
Breaks all the paradigms of the good codebase!

File: frontend/src/components/Course/Course.tsx [hunk 3]
```diff
@@ -13,25 +15,61 @@ import {
...
+          <MobileCourseCompleted>
+            Completed <img alt="" src={checkIcon} />
+          </MobileCourseCompleted>
+        )}
+      </CourseTextContainer>
+      {pageName === PAGES.myCourses && status !== 'completed' && (
```
Comment (k-krasilnikova) [inline frontend/src/components/Course/Course.tsx:68]: !

File: frontend/src/components/Course/Course.tsx [hunk 4]
```diff
@@ -13,25 +15,61 @@ import {
...
+        <MobileCourseProgress>
+          <ProgressBar value={progress} text={`${progress}%`} size="medium" textColor="#000000" />
```
Comment (k-krasilnikova) [inline frontend/src/components/Course/Course.tsx:70]: Move please color to the progressBar theme

**Review by k-krasilnikova — CHANGES_REQUESTED** [review]

File: frontend/src/components/Course/Course.tsx (hunk 2 above)
Comment (Julia-Maliutsina) [reply frontend/src/components/Course/Course.tsx]: Moved it to constants

File: frontend/src/components/Course/Course.tsx [hunk 5]
```diff
@@ -13,25 +16,66 @@ import {
...
+          <ProgressBar
+            value={progress}
+            text={`${progress}%`}
+            size="medium"
+            variant="mobileCourse"
+          />
+        </MobileCourseProgress>
+      )}
     </AboutCourseContainer>
     <ButtonsContainer>
```
Comment (NikitaKhadnevich) [inline frontend/src/components/Course/Course.tsx:80]: We have problem with position button block  like on this screen (https://radikal.ru/lfp/b.radikal.ru/b23/2202/a4/b5b1e3854eef.png/htm)
https://a.radikal.ru/a29/2202/93/1f982dbf23ef.png

File: frontend/src/pages/MyCourses/MyCoursesList.tsx [hunk 6]
```diff
@@ -24,16 +25,19 @@ const MyCoursesList: React.FC<ResponseDataMyCourses> = ({ data, isLoading }) =>
...
                 title={object.course.title}
                 description={object.course.description}
                 duration={object.course.duration}
                 lessons={object.course.lessons}
+                pageName="myCourses"
+                status={object.status}
+                progress={countProgress(object.progress)}
               >
                 <CourseActionsBox>
                   <CourseActions>
```
Comment (NikitaKhadnevich) [inline frontend/src/pages/MyCourses/MyCoursesList.tsx:43]: There is some error in console which includes next wrong: 
`Check the render method of `MyCoursesList`. See https://reactjs.org/link/warning-keys for more information.`

**Review by NikitaKhadnevich — CHANGES_REQUESTED** [review]
We had lost menu arrow click area again)Its work only on current arrow

File: frontend/src/pages/MyCourses/MyCoursesList.tsx (hunk 6 above)
Comment (Julia-Maliutsina) [reply frontend/src/pages/MyCourses/MyCoursesList.tsx:43]: This was fixed in develop after I created this branch

**Comment by Julia-Maliutsina** [comment]
> We had lost menu arrow click area again)Its work only on current arrow

I didn't touch menu arrow in this branch. Problems with arrow are already fixed in develop and this branch was just created earlier

**Review by k-krasilnikova — APPROVED** [review]

**Pull request merged**